src/storage/mod.rs

```diff
@@ -1308,28 +1298,28 @@ pub struct TestStorageBuilder<E: Engine, L: LockManager> {
     engine: E,
     config: Config,
     pipelined_pessimistic_lock: bool,
-    lock_mgr: Option<L>,
+    lock_mgr: L,
```

Comment:
It's better to use `DummyLockManager` as the default one and provide an interface to change it, so that you don't need to change many codes. For example,
```rust
pub struct TestStorageBuilder<E: Engine, L: LockManager> {
...
    lock_mgr: Option<L>,
}

impl<E: Engine, L: LockManager> TestStorageBuilder<E, L> {	
    pub fn set_lock_mgr(mut self, lock_mgr: L) -> Self {	
        self.lock_mgr = Some(lock_mgr);	
        self	
    }

    pub fn build(self) -> Result<Storage<E, L>> {
        let read_pool = build_read_pool_for_test(
            &crate::config::StorageReadPoolConfig::default_for_test(),
            self.engine.clone(),
        );

        Storage::from_engine(
            self.engine,
            &self.config,
            ReadPool::from(read_pool).handle(),
            self.lock_mgr.unwrap_or_else(|| DummyLockManager{}),
            self.pipelined_pessimistic_lock,
        )
    }
}
```

Reply:
Sg, let me check

Reply:
One q though is whether we still want to bring back the Option<L>. As the downstream always expects a valid lock manager, making it non optional will ensure it gets defined.

Reply:
In `build`, if `lock_mgr` is none, use `DummyLockManager` otherwise the provided one.
```rust
            self.lock_mgr.unwrap_or_else(|| DummyLockManager{}),
```